OPAL/ide/src/main/scala/org/opalj/ide/solver/IDEAnalysis.scala

```diff
@@ -0,0 +1,1115 @@
+        val cs = s.getAndRemoveDependeeContinuations(eps)
+
+        // Call continuations
+        cs.foreach(c => c())
+
+        // The continuations can have enqueued paths to the path work list
+        continuePhase1()
+        continuePhase2()
+
+        createResult()
+    }
+
+    private def seedPhase1()(implicit s: State): Unit = {
+        def propagateSeed(e: Path, callable: Callable, f: EdgeFunction[Value]): Unit = {
+            val oldJumpFunction = s.getJumpFunction(e)
+            val fPrime = f.meetWith(oldJumpFunction)
+
+            if (!fPrime.equalTo(oldJumpFunction)) {
+                s.setJumpFunction(e, fPrime)
+                s.enqueuePath(e)
+                s.rememberCallableWithChanges(callable)
+            }
+        }
+
+        val callables = s.getTargetCallables
+        callables.foreach { callable =>
+            // IDE P1 lines 5 - 6
+            icfg.getStartStatements(callable).foreach { stmt =>
+                val path = ((stmt, problem.nullFact), (stmt, problem.nullFact))
+                propagateSeed(path, callable, identityEdgeFunction)
+
+                problem.getAdditionalSeeds(stmt, callable).foreach { fact =>
+                    val path = ((stmt, problem.nullFact), (stmt, fact))
+                    propagateSeed(path, callable, identityEdgeFunction)
+
+                    def processAdditionalSeed(): Unit = {
+                        val edgeFunction =
+                            handleEdgeFunctionResult(
+                                problem.getAdditionalSeedsEdgeFunction(stmt, fact, callable),
+                                processAdditionalSeed _
+                            )
+                        propagateSeed(path, callable, edgeFunction)
+                    }
+
+                    processAdditionalSeed()
+                }
+            }
+        }
+
+        logDebug(s"seeded with ${s.getPathWorkListSize} path(s)")
+    }
+
+    private def processPathWorkList()(implicit s: State): Unit = {
+        while (!s.isPathWorkListEmpty) { // IDE P1 line 7
+            val path = s.dequeuePath() // IDE P1 line 8
+            val ((_, _), (n, _)) = path
+            val f = s.getJumpFunction(path) // IDE P1 line 9
+
+            logDebug("\nprocessing next path")
+            logTrace(s"path=${pathToString(path)}")
+            logTrace(s"current jumpFunction=$f")
+
+            if (icfg.isCallStatement(n)) { // IDE P1 line 11
+                processCallFlow(path, f, icfg.getCallees(n))
+            } else if (icfg.isNormalExitStatement(n)) { // IDE P1 line 19
+                processExitFlow(path, f)
+            } else { // IDE P1 line 30
+                processNormalFlow(path, f)
+            }
+
+            logDebug(s"${s.getPathWorkListSize} path(s) remaining after processing last path")
+        }
+    }
+
+    private def processCallFlow(path: Path, f: JumpFunction, qs: collection.Set[Callable])(
```

Comment:
This code is poorly documented and uses many nondescriptives names. What are things like `sp`, `n`, `rs`, `qs`, `d5`, `d5s`, `d3`, `sqs`, etc.